resources/default.rb

```diff
@@ -4,19 +4,59 @@
 property :artifactoryonline, :kind_of => String
 property :repository, :kind_of => String, :required => true
 property :repository_path, :kind_of => String, :required => true
+property :artifact_name, :kind_of => String, :required => true
 property :artifactory_username, :kind_of => String
 property :artifactory_password, :kind_of => String
+property :highest, :kind_of => [TrueClass, FalseClass]
 
 property :group, :kind_of => [Integer, String]
 property :mode, :kind_of => [Integer, String]
 property :owner, :kind_of => [Integer, String]
 
 action_class do
   require "uri"
+  require 'net/http'
+  require 'json'
 
   include ::ArtifactoryArtifact::Helper
 
+  def fetch_highest(new_resource)
+    # Fetching list of artifacts from artifactory
+    url = URI("#{new_resource.artifactory_url}/api/search/aql/")
+    http = Net::HTTP.new(url.host, url.port)
+    request = Net::HTTP::Post.new(url)
+
+    request['Content-Type'] = 'text/plain'
+    request['Accept'] = '*/*'
+    request['Host'] = url.host
+    request['Connection'] = 'keep-alive'
+    request.basic_auth new_resource.artifactory_username, new_resource.artifactory_password
+    request.body = "items.find({\"repo\":{\"$eq\":\"   #{new_resource.repository}\"}})"
+    response = http.request(request)
+
+    parsed_json = JSON.parse(response.body)
+    name_re = new_resource.artifact_name.gsub(/HIGHEST/, '([\.0-9]+)')
+
+    # Adding Version tag to parsed_json
+    parsed_json['results'].each do |res|
+      output = res['name']
+      output =~ /^#{name_re}$/
+      res['version'] = $1
+    end
+
+    #Sorting versions and picking highest value
+    parsed_json['results'].sort! { |x,y|
+      Chef::Provider::Package::Yum::RPMUtils.rpmvercmp(x['version'], y['version'])
```

Comment:
Hmm, relying on RPM specific feature to sort versions might not be a good idea.... At very least I want it to support Debian/Ubuntu.

Would't it be possible to pick single latest version by Artifactory...? If it is possible, we would not need to sort version locally.

Reply:
rpmvercmp was selected because it's one of the most generic vercmp's that supports mixed alpha-numeric versions (like `1.2a`) and it's included in Chef.  Since it's based on the same open-source upstream software version requirements, it should be virtually equivalent to apt except in possibly edge conditions (in which case we'd need to have plug-ins per repo-type).  It also happens to support generic, common, or semver well, which is also why it was used.

Surprisingly, Artifactory does not have a "highest version" sort.  Artifactory _only_ has a "latest version" sort, which is based on upload-time, not version number.  Thus a recently uploaded patch fix 1.0.2 on an older release can sort "later" than an already released 2.1.